src/components/victory-chart.jsx

```diff
@@ -881,157 +583,89 @@ export default class VictoryChart extends React.Component {
-    const lineStyle = {data: _.merge({}, this.style.line.data, attrs)};
-    const style = _.merge({}, {parent: this.style.parent}, this.style.line, lineStyle);
-    return (
-      <VictoryLine
-        {...this.props}
-        animate={animate}
-        standalone={false}
-        data={dataset.data}
-        label={attrs.label}
-        interpolation={attrs.interpolation || this.props.interpolation}
-        style={style}
-        domain={this.domain}
-        range={this.range}
-        ref={name}
-        key={"line-" + dataset.index}/>
-    );
-  }
-
-  drawScatter(dataset) {
-    const animate = this.props.animate && (this.props.animate.scatter || this.props.animate);
-    const {type, name, symbol, size, ...attrs} = dataset.attrs;
-    const scatterStyle = {data: _.merge({}, this.style.scatter.data, attrs)};
-    const style = _.merge({}, {parent: this.style.parent}, this.style.scatter, scatterStyle);
-    return (
-      <VictoryScatter
-        {...this.props}
-        animate={animate}
-        standalone={false}
-        data={dataset.data}
-        size={size || 3}
-        symbol={symbol || "circle"}
-        style={style}
-        domain={this.domain}
-        range={this.range}
-        ref={name}
-        key={"scatter-" + dataset.index}/>
-    );
-  }
-
-  drawBars(datasets, options) {
-    const animate = this.props.animate && (this.props.animate.bar || this.props.animate);
-    const categories = (this.stringMap.x) && _.keys(this.stringMap.x);
-    return (
-      <VictoryBar
-        {...this.props}
-        animate={animate}
-        standalone={false}
-        data={_.pluck(datasets, "data")}
-        dataAttributes={_.pluck(datasets, "attrs")}
-        stacked={(options && !!options.stacked) ? options.stacked : false}
-        style={_.merge({}, {parent: this.style.parent}, this.style.bar)}
-        domain={this.domain}
-        range={this.range}
-        categories={this.props.categories || categories}
-        categoryLabels={this.props.categoryLabels}
-        key={"bar"}/>
-    );
-  }
-
-  drawStackedBars(datasets) {
-    return this.drawBars(datasets, {stacked: true});
+  generateData(child) {
+    if (!child.props.y) {
+      return undefined;
+    }
+    const generateX = (component) => {
+      const domain = this.domain.x;
+      const samples = _.isArray(component.props.y) ? component.props.y.length : 50;
+      const step = _.max(domain) / samples;
+      // return an array of x values spaced across the domain,
+      // include the maximum of the domain
+      return _.union(_.range(_.min(domain), _.max(domain), step), [_.max(domain)]);
+    };
+    const y = child.props.y;
+    const xArray = child.props.x || generateX(child);
+    if (_.isFunction(y)) {
+      return _.map(xArray, (x) => {
+        return {x, y: y(x)};
+      });
+    } else {
+      const n = _.min([xArray.length, y.length]);
+      return _.map(_.take(xArray, n), (x, index) => {
+        return { x, y: y[index]};
+      });
+    }
```

Comment:
I find all these small functions and nested lodash calls hard to follow. Can we break some of these out into `util.js` so this functionality has a name and clear parameters?